src/Sylius/Behat/Context/Ui/Admin/ManagingPaymentMethodsContext.php

```diff
@@ -0,0 +1,373 @@
+            sprintf('Payment method with name %s has not been found.', $paymentMethodName)
+        );
+    }
+
+    /**
+     * @When /^I want to browse payment methods$/
+     */
+    public function iWantToBrowsePaymentMethods()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @Then I should see :amount payment methods in the list
+     */
+    public function iShouldSeePaymentMethodsInTheList($amount)
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::true(
+            ((int) $amount) === $foundRows,
+            sprintf('%s rows with payment methods should appear on page, %s rows has been found', $amount, $foundRows)
+        );
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatIsRequired($element)
+    {
+        $this->assertFieldValidationMessage($element, sprintf('Please enter payment method %s.', $element));
+    }
+
+    /**
+     * @Then the payment method with :element :value should not be added
+     */
+    public function thePaymentMethodWithElementValueShouldNotBeAdded($element, $value)
+    {
+        $this->iWantToBrowsePaymentMethods();
+
+        Assert::false(
+            $this->indexPage->isResourceOnPage([$element => $value]),
+            sprintf('Payment method with %s %s was created, but it should not.', $element, $value)
+        );
+    }
+
+    /**
+     * @Then /^(this payment method) should still be named "([^"]+)"$/
+     */
+    public function thisShippingMethodNameShouldBe(PaymentMethodInterface $paymentMethod, $paymentMethodName)
+    {
+        $this->iWantToBrowsePaymentMethods();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage(
+                [
+                    'code' => $paymentMethod->getCode(),
+                    'name' => $paymentMethodName,
+                ]
+            ),
+            sprintf('Payment method name %s has not been assigned properly.', $paymentMethodName)
+        );
+    }
+
+    /**
+     * @param string $element
+     * @param string $expectedMessage
+     */
+    private function assertFieldValidationMessage($element, $expectedMessage)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->createPage, $this->updatePage);
+
+        Assert::true(
+            $currentPage->checkValidationMessageFor($element, $expectedMessage),
+            sprintf('Payment method %s should be required.', $element)
+        );
+    }
+
+    /**
+     * @Then /^the code field should be disabled$/
+     */
+    public function theCodeFieldShouldBeDisabled()
+    {
+        Assert::true(
+            $this->updatePage->isCodeDisabled(),
+            'Code field should be disabled'
+        );
+    }
+
+    /**
+     * @Then this payment method should be enabled
+     */
+    public function thisCountryShouldBeEnabled()
+    {
+        Assert::true(
+            $this->updatePage->isPaymentMethodEnabled(),
```

Comment:
Can't we make this assertion on index page? I think it is worthy to tell user is it enabled or not when one would browse the products